packages/axelar-soroban-std/src/error.rs

```diff
@@ -104,3 +104,33 @@ macro_rules! assert_some {
         }
     };
 }
+
+#[macro_export]
+macro_rules! assert_invoke_auth {
+    ($caller:expr, $client:ident . $method:ident ( $($arg:expr),* $(,)? )) => {{
+        use soroban_sdk::xdr::{ScError, ScErrorCode, ScVal};
+
+        let call_result = $client
+            .mock_auths(&[MockAuth {
+                address: &$caller,
+                invoke: &MockAuthInvoke {
+                    contract: &$client.address,
+                    fn_name: stringify!($method),
+                    args: ($($arg.clone(),)*).into_val(&$client.env),
+                    sub_invokes: &[],
+                },
+            }])
+            .$method($($arg),*);
```

Comment:
```suggestion
            .try_$method($($arg),*);
```
can you add the `try` part in the macro, so the usage just looks like `client.rotate_signers(..)`

Reply:
I’ve tried different approaches, but I’m not sure this will work. Due to the logic comparing the `call_result` to `Err`, it’s producing an error.

```
match call_result {
            Err(_) => {
            }
            Ok(_) => panic!("Expected error, but got Ok result."),
        }
```

Reply:
I’ve tried different approaches, but I’m not sure this will work. Due to the logic matching the `call_result` to `Err`, it’s producing an error if I try to use `try_$method`.